2-0-data-structures-and-algorithms/src/test/java/com/bobocode/linked_queue/LinkedQueueTest.java

```diff
@@ -1,103 +1,372 @@
+        assertThat(hasNodeField).isTrue();
     }
 
     @Test
     @Order(3)
-    void addElementIntoEmptyQueue() {
-        integerQueue.add(1312);
+    void checkFieldsInQueueCLass() {
+        Class<?> baseClass = this.integerQueue.getClass();
+        boolean hasHeadField = hasField(baseClass, HEAD_FIELD);
+        boolean hasSizeFiled = hasField(baseClass, SIZE_FIELD);
+        boolean hasTailFiled = hasField(baseClass, TAIL_FIELD);
 
-        assertEquals(1, integerQueue.size());
-        assertEquals(1312, integerQueue.poll().intValue());
+        assertThat(hasHeadField).isTrue();
+        assertThat(hasSizeFiled).isTrue();
+        assertThat(hasTailFiled).isTrue();
     }
 
     @Test
     @Order(4)
-    void addElement() {
-        integerQueue.add(324);
-        integerQueue.add(23);
-        integerQueue.add(5);
+    void addFillsQueueWhenItIsEmpty() {
+        integerQueue.add(1);
+        integerQueue.add(228);
+        int size = getInternalSize();
+        boolean isEmpty = isEmptyQueue();
+        Integer firstElement = (Integer) pollElementFromQueue();
+        Integer secondElement = (Integer) pollElementFromQueue();
 
-        assertEquals(3, integerQueue.size());
-        assertEquals(324, integerQueue.poll().intValue());
+        assertThat(size).isEqualTo(2);
+        assertThat(isEmpty).isEqualTo(false);
+        assertThat(firstElement).isEqualTo(1);
+        assertThat(secondElement).isEqualTo(228);
     }
 
     @Test
     @Order(5)
-    void pollElementFromEmptyQueue() {
-        assertNull(integerQueue.poll());
+    void addFillsQueueWhenItIsNotEmpty() {
+        addIntElementToQueue(12);
+        integerQueue.add(111);
+        int size = getInternalSize();
+        boolean isEmpty = isEmptyQueue();
+        Integer firstElement = (Integer) pollElementFromQueue();
+        Integer secondElement = (Integer) pollElementFromQueue();
+
+        assertThat(size).isEqualTo(2);
+        assertThat(isEmpty).isEqualTo(false);
+        assertThat(firstElement).isEqualTo(12);
+        assertThat(secondElement).isEqualTo(111);
     }
 
     @Test
     @Order(6)
-    void pollElement() {
-        integerQueue.add(33);
-        integerQueue.add(123);
-        integerQueue.add(222);
-        integerQueue.add(444);
-
-        integerQueue.poll(); // should poll 33
+    void addIncreasesQueueSize() {
+        integerQueue.add(1);
+        integerQueue.add(2);
+        int size = this.integerQueue.size();
 
-        assertEquals(123, integerQueue.poll().intValue());
-        assertEquals(2, integerQueue.size());
+        assertThat(size).isEqualTo(2);
     }
 
     @Test
     @Order(7)
-    void pollLastElement() {
-        integerQueue.add(8);
-        integerQueue.add(123);
-        integerQueue.add(99);
-        integerQueue.add(46);
-
-        integerQueue.poll(); // should poll 8
-        integerQueue.poll(); // should poll 123
-        integerQueue.poll(); // should poll 99
+    void isEmptyReturnsFalseWhenElementIsAdded() {
```

Comment:
We should avoid using of several implemented methods. Lets remain only `isEmpty()`, replace adding by reflection, name the test `isEmptyReturnsFalseWhenQueueHasElement()` and place it in the end of the order.